src/Classes/Utils/CanvasUtils.ts

```diff
@@ -0,0 +1,348 @@
+            case ShapeType.character:
+                const style = particle.container.options.particles.shape.character.style;
+                const weight = particle.container.options.particles.shape.character.weight;
+                const size = Math.round(radius) * 2;
+                const font = particle.container.options.particles.shape.character.font;
+                const text = particle.text;
+
+                context.font = `${style} ${weight} ${size}px ${font}`;
+
+                if (text) {
+                    const pos = {
+                        x: -radius / 2,
+                        y: radius / 2
+                    };
+
+                    CanvasUtils.drawTextShape(context, text, pos, particle.container.options.particles.shape.character.fill);
+                }
+                break;
+
+            case ShapeType.image:
+                if (particle.image && particle.image.data.obj) {
+                    CanvasUtils.drawImageShape(context, particle, radius);
+                }
+
+                break;
+        }
+    }
+
+    private static drawTriangleShape(context: CanvasRenderingContext2D, radius: number) {
+        const start: ICoordinates = {
+            x: -radius,
+            y: radius / 1.66,
+        };
+
+        const side: ISide = {
+            count: {
+                denominator: 2,
+                numerator: 3,
+            },
+            length: radius * 2,
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawPolygonShape(context: CanvasRenderingContext2D, radius: number, sides: number) {
+        const start: ICoordinates = {
+            x: -radius / (sides / 3.5),
+            y: -radius / (2.66 / 3.5),
+        };
+        const side: ISide = {
+            count: {
+                denominator: 1,
+                numerator: sides,
+            },
+            length: radius * 2.66 / (sides / 3),
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawStarShape(context: CanvasRenderingContext2D, radius: number, sides: number) {
+        const start: ICoordinates = {
+            x: -radius * 2 / (sides / 4),
+            y: -radius / (2 * 2.66 / 3.5),
+        };
+        const side: ISide = {
+            count: {
+                denominator: 2,
+                numerator: sides,
+            },
+            length: radius * 2 * 2.66 / (sides / 3),
+        };
+
+        CanvasUtils.drawGenericPolygonShape(context, start, side);
+    }
+
+    private static drawLineShape(context: CanvasRenderingContext2D, length: number, stroke: IStroke) {
+        context.moveTo(0, -length/2);
+        context.lineTo(0, length / 2);
+        context.strokeStyle = stroke.color;
+        context.lineWidth = stroke.width;
+        context.stroke();
+    }
+
+    private static drawCircleShape(context: CanvasRenderingContext2D, radius: number, center: ICoordinates) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'drawCircleShape' to have a typedef](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)